### PR TITLE
dts: arm: nordic: Define power states for nrf54h20/cpuapp

### PR DESCRIPTION
**Pull request by nordic-segl**
Add definition of low power states 'idle' and 's2ram' for nrf54h20/cpuapp.

### CODE REVIEW

**Comment by nordic-segl** [comment]
FYI: @nordic-piks @adamkondraciuk @nordic-krch @gmarull

File: dts/arm/nordic/nrf54h20_cpuapp.dtsi [hunk 1]
```diff
@@ -59,3 +59,23 @@ wdt011: &cpuapp_wdt011 {};
 &grtc {
 	interrupts = <109 NRF_DEFAULT_IRQ_PRIORITY>;
 };
+
+/ {
+	power-states {
```
Comment (gmarull) [inline dts/arm/nordic/nrf54h20_cpuapp.dtsi:64]: power states must be within cpu node, check other cases

Comment (nordic-segl) [reply dts/arm/nordic/nrf54h20_cpuapp.dtsi:64]: Changed from root node to cpus like here
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/silabs/efr32bg2x.dtsi#L52
Is this what You meant?

**Comment by nordic-segl** [comment]
devicetree error: [path_to]/zephyr/dts/arm/nordic/nrf54h20_cpuapp.dtsi:61 (column 1): parse error: undefined node label 'cpus'

**Comment by nordic-segl** [comment]
> devicetree error: [path_to]/zephyr/dts/arm/nordic/nrf54h20_cpuapp.dtsi:61 (column 1): parse error: undefined node label 'cpus'

Fixed

**Review by gmarull — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**